### ISSUE TITLE
Avatar resize overlap on cell

### ISSUE THREAD

**Issue by aaasifrz9** [MessageKit/MessageKit#536]
## General Information

I tried to resize avatar with below code and it resize the avatar but overlap ob cell. Also, it shifted to bottom from baseline of cell. Right avatar is hiding.

```swift
func configureAvatarView(_ avatarView: AvatarView, for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) {
    let avatar = SampleData.shared.getAvatarFor(sender: message.sender)
    avatarView.set(avatar: avatar)

    avatarView.frame = CGRect(x: avatarView.frame.origin.x, y: avatarView.frame.origin.y, width: 50.0, height: 50.0)
 }
```

Link Sample: https://github.com/MessageKit/MessageKit
Page: ConversationViewController.swift
Method: configureAvatarView in extension ConversationViewController: MessagesDisplayDelegate

Also if required to set avatar vertically centre aligned to the cell. How can we?

Please help. Thanks in advance.

@SD10 ,@nathantannar4 , @zhongwuzw

![img_1841](https://user-images.githubusercontent.com/24565054/36578532-d0847d34-1883-11e8-9512-8666256bea19.PNG)

**Comment by SD10** [comment]
@aaasifrz9 It's likely because you are using this method to modify the `AvatarView`'s size. You need to make sure that this value lines up with `avatarSize(for:at:in)` method of `MessagesLayoutDelegate`. I wouldn't recommend changing the avatar's size or frame in this method. Only things like setting the image async or setting the view to hidden.

**Comment by aaasifrz9** [comment]
Thanks for quick response. I'll check.

**Comment by aaasifrz9** [comment]
Can we resize Avatar? If yes, where can we set avatar frame? Please help.

**Comment by aaasifrz9** [comment]
Thanks a lot @SD10 , avatarSize(for:at:in) is working properly.

**Comment by aaasifrz9** [comment]
Thanks again but when I tried to resize avatar using avatarSize(for:at:in) with some random size it gives me result like below screenshots. It overlaps with date/avatar. Sizes are 100 and 150 using iPhone 6.

![img_1843](https://user-images.githubusercontent.com/24565054/36578884-0656ab60-1886-11e8-9003-3745b36b207a.PNG)
![img_1842](https://user-images.githubusercontent.com/24565054/36578885-06928234-1886-11e8-993f-ff40efc83fa6.PNG)

**Comment by nathantannar4** [comment]
I guess we never intended that avatars would be that big...

**Comment by aaasifrz9** [comment]
Then, how to do something like below example.

If possible, please help.

![screen shot 2018-02-23 at 10 51 25 am](https://user-images.githubusercontent.com/24565054/36579167-acacb8aa-1887-11e8-9592-11933787bae1.png)

**Comment by SD10** [comment]
@aaasifrz9 Are you still moving the AvatarView using?

`func configureAvatarView(_ avatarView: AvatarView, for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) {}`

**Comment by aaasifrz9** [comment]
No, above sample is created in photoshop. I need to do something like this.

**Comment by aaasifrz9** [comment]
Hi @SD10 , I have set vertically centre align using below code with Avatar resize 80 using `avatarSize(for:at:in)` as mentioned below: -

```
func avatarPosition(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> AvatarPosition {
    return AvatarPosition(horizontal: .natural, vertical: .messageCenter)
}
```

Now sometimes I can see `cellBottomLabelAttributedText` (Display date) is showing left user differently with left aligned also right aligned with text cell b bubble. This also happens for right users. And overlaps at avatar.

[Default Padding.zip](https://github.com/MessageKit/MessageKit/files/1750615/Default.Padding.zip)


Then I changed margin 50 it shows me overlap issue with avatar attached, please help: -

```
func cellBottomLabelAlignment(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> LabelAlignment {
    if isFromCurrentSender(message: message) {
        return .messageLeading(UIEdgeInsets(top: 0, left: 50, bottom: 0, right: 0))
    } else {
        return .messageTrailing(UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 50))
    }
}
```

[Padding 50.zip](https://github.com/MessageKit/MessageKit/files/1750617/Padding.50.zip)

Please help, how to maintain date padding properly for each cell.

**Comment by SD10** [comment]
@aaasifrz9 I think I've fixed this in MessageKit 1.0.0-beta.1. Thanks for reporting 👍 Feel free to reopen this issue if the problem persists

**Comment by justdan0227** [comment]
So what it the correct way to resize the Avatar Image?
